src/Microsoft.VisualStudio.Editor.Razor/BraceSmartIndenter.cs

```diff
@@ -242,35 +242,73 @@ internal static void RestoreCaretTo(int caretPosition, ITextView textView)
         }
 
         // Internal for testing
-        internal static bool AtValidContentKind(int changePosition, RazorSyntaxTree syntaxTree)
+        internal static bool AtApplicableRazorBlock(int changePosition, RazorSyntaxTree syntaxTree)
         {
+            // Our goal here is to return true when we're acting on code blocks that have all 
+            // whitespace content and are surrounded by metacode.
+            // Some examples:
+            // @functions { |}
+            // @section foo { |}
+            // @{ |}
+
             var change = new SourceChange(changePosition, 0, string.Empty);
             var owner = syntaxTree.Root.LocateOwner(change);
 
-            if (owner == null)
+            if (IsUnlinkedSpan(owner))
             {
                 return false;
             }
 
-            if (owner.Kind == SpanKindInternal.MetaCode)
+            if (SurroundedByInvalidContent(owner))
             {
-                // @functions{|}
-                return true;
+                return false;
             }
 
-            if (owner.Kind == SpanKindInternal.Code)
+            if (ContainsInvalidContent(owner))
+            {
+                return false;
+            }
+
+            // Indentable content inside of a code block.
+            return true;
+        }
+
+        // Internal for testing
+        internal static bool ContainsInvalidContent(Span owner)
+        {
+            if (owner.Symbols.Count == 0)
```

Comment:
Why was this necessary? Why not indent empty code blocks as well? Please add a comment.

Reply:
Totally forget, gonna have to run through the scenarios again to re-verify.

Reply:
Has 0 value. Good catch 👍 👍 👍